### ISSUE TITLE
mlx.core.conv1d/conv2d exceptions when padding is negative

### ISSUE THREAD

**Issue by Redmept1on** [ml-explore/mlx#548]
**Describe the bug**
1. abort appears when mlx.core.conv1d padding is set to negative, indicating that there is a lack of padding checks, and I don't think mlx should be trying to allocate memory fundamentally.
2. mlx.core.conv2d returns an empty array when the padding is set to negative, and there is no crash.

Below with the torch.nn.functional.conv1d/conv2d comparisons.

**To Reproduce**

Conv1d
```python
import mlx.core as mx

input_array = mx.array([[[1., 2, 3], [4, 5, 6]]])  
weight_array = mx.array([[[1., 2,3], [4, 5, 6]]]) 

stride = 1
padding = -1
dilation = 1
groups = 1 

conv1d = mx.conv1d(input_array, weight_array, stride=stride, padding=padding, dilation=dilation, groups=groups)
print(conv1d)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/1e193000-bb52-4895-9da9-7b3c97ac8dec)

```python
import torch
import torch.nn.functional as F

input_tensor = torch.tensor([[[1., 2, 3], [4, 5, 6]]]) 
weight_tensor = torch.tensor([[[1., 2, 3], [4, 5, 6]]])  

stride = 1
padding = -1
dilation = 1
groups = 1

conv1d_result = F.conv1d(input_tensor, weight_tensor, stride=stride, padding=padding, dilation=dilation, groups=groups)
print("Conv1d result:", conv1d_result)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/7c54a0e7-5683-4602-9b72-b17fd2557323)

Conv2d
```python
import mlx.core as mx

input_array = mx.array([[[[1., 2], [3, 4]]]]) 
weight_array = mx.array([[[[1, 0.5], [0.5, 1]]]]) 

stride = 1
padding = -1
dilation = 1
groups = 1  

conv2d = mx.conv2d(input_array, weight_array, stride=stride, padding=padding, dilation=dilation, groups=groups)
print(conv2d)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/d39f42b9-6c13-40ea-a7c2-6762f520753b)

```python
import torch
import torch.nn.functional as F

input_tensor = torch.tensor([[[[1., 2], [3, 4]]]]) 
weight_tensor = torch.tensor([[[[1, 0.5], [0.5, 1]]]])  

stride = 1
padding = -1
dilation = 1
groups = 1

conv2d_result = F.conv2d(input_tensor, weight_tensor, stride=stride, padding=padding, dilation=dilation, groups=groups)
print("Conv2d result:", conv2d_result)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/5e3b92bf-72d7-4df9-9787-9f82dbdecd51)

**Expected behavior**
padding may need to add positive and negative checks.

**Desktop (please complete the following information):**
 - OS Version: MacOS 14.2.1
 - Version 0.10.0
 - torch 2.0.0+cu117

**Comment by jagrit06** [comment]
Thanks for taking the time to bring this to our attention. I will be closing the issue for now.

**Comment by Redmept1on** [comment]
Thank you for your patience in dealing with the questions I submitted! May I ask if issue547 can be regarded as enhancenemt and issue548 can be regarded as a bug? @awni

**Comment by awni** [comment]
I agree with your assessment re bug for #548 and enhancement for #547.

**Comment by Redmept1on** [comment]
thx！
